Report/Chapters/Design.tex

```diff
@@ -1,17 +1,103 @@
+This makes using the program on a large multi-touch table easier.
+However, the implementation of the menu would take too much time and the idea was therefore dismissed.
+
+The final design has two separate menus, one for adding new blocks and one for options on existing blocks.
+
+The primary menu uses `drawers' for the categories of the functions.
+This way it is easy to find functions even without an extensive knowledge of Haskell.
+In- and ontput blocks and the function definition block can be created using buttons at the bottom of the menu.
+This location is chosen to make them quickly available.
+The primary menu can be opened multiple times at once and moved around the screen so multiple users can work on a single program.
+
+The context menu is compact so is does not take up too much space.
+It contains clear icons which represent different actions.
+The icons are large enough to touch, even for people with large fingers.
+
+\subsubsection{Color scheme}
+
+The background color is a dark gray.
+This is chosen because it is less tiring for the eyes than white and allows for a high contrast.
+The background also incorporates a grid pattern so users can align elements on the screen if they like to.
+
+All other components feature a cheerful color scheme.
+This color scheme provides good contrast with the background and good recognizability for the different kinds of blocks.
+
+\subsection{Behaviour and interaction design}
+
 \section{Languages and libraries}
 
-% Java. Why?
-% JavaFX. Why?
-% TactileFX. Why?
+Viskell is written in Java using Java 8 with JavaFX.
+Java 8 is chosen because it, unlike Java 7, still receives public updates at the moment of writing.
+Language level 8 is chosen to make use of some of the new features in this version.
+
+Viskell uses TactileFX, a JavaFX touch screen library developed at the University of Twente.
+The client wanted the program to use this library, which also means that Viskell had to use Java and JavaFX.
+
+Furthermore, the following libraries have been used:
+
+\begin{itemize}
+	\item Antlr 4 (http://www.antlr.org/)
+	\item Guava (https://github.com/google/guava)
+	\item JFXtras (http://jfxtras.org/)
+\end{itemize}
 
 \section{High-level architecture}
 
-% Frontend/backend split
-% Communication between frontend and backend (when we've got that settled)
+Viskell consists of two parts, a front end and a back end.
+The front end is responsible for providing a user interface, handling user input, displaying error messages, and communicating with the back end.
+The back end consists of a representation for Haskell code in Java, an interface to GHCi and functionality to make working with the back end easier.
+This separation is made to make development and debugging for the part that generates Haskell code easier.
+It also makes using the user interface for a different purpose a bit easier, although the user interface is not specifically designed to be reusable.
```

Comment:
Or, to put it more positively, the user interface has been specifically designed for our use case.